packages/react-query/src/useBaseQuery.ts

```diff
@@ -102,10 +102,10 @@ export function useBaseQuery<
 
   // Handle suspense
   if (
-    defaultedOptions.suspense &&
-    result.isLoading &&
-    result.isFetching &&
-    !isRestoring
+    !isRestoring &&
+    (typeof defaultedOptions.suspense === 'function'
+      ? defaultedOptions.suspense(observer.getCurrentQuery())
+      : defaultedOptions.suspense && result.isLoading && result.isFetching)
```

Comment:
can you please add a test for when `suspense` is a function?

there is also the thing that `useErrorBoundary` is set to `true` if we have suspense set:

https://github.com/TanStack/query/blob/454c58481f6f5d5c3e182e6e418d65fdb4a9ec1f/packages/query-core/src/queryClient.ts#L681-L683

so with that change, `useErrorBoundary` will be `true` when a function is passed to `suspense`. I'm not sure that this is right?

Reply:
It's fine, cuz the throw promise behavior of `useBaseQuery`  depend not only on suspense, and still depend on `isLoading` and `isFetching`. If u wanna put the suspend function on suspense, it's unavoidable.

and I think,  If people set `suspense` to function, It shoud be a metaphor that people will wanna this behavior like set it to `true`.